utils/dom.js

```diff
@@ -150,21 +129,19 @@ export function getRectangleFromRange( range ) {
 		return range.getBoundingClientRect();
 	}
 
-	// If the collapsed range starts (and therefore ends) at an element node,
-	// `getClientRects` will return undefined. To fix this we can get the
-	// bounding rectangle of the element node to create a DOMRect based on that.
-	if ( range.startContainer.nodeType === ELEMENT_NODE ) {
-		const { x, y, height } = range.startContainer.getBoundingClientRect();
+	let rect = range.getClientRects()[ 0 ];
 
-		// Create a new DOMRect with zero width.
-		return new DOMRect( x, y, 0, height );
+	// If the collapsed range starts (and therefore ends) at an element node,
+	// `getClientRects` can be empty in some browsers. This can be resolved
+	// by adding a temporary text node to the range.
+	if ( ! rect ) {
+		const padNode = document.createTextNode( '\u200b' );
```

Comment:
If this works for all cases in your testing, then cool, let's do it like this and I'll ponder there could be any better solution.